Cohorted Courseware carol/docs/CohortDOC-1222

catong opened the pull request:
@explorerleslie @andy-armstrong @cahrens @dan-f @talbs @jimabramson 
@mhoeber @lamagnifica @srpearce 
This is draft documentation for the cohorted courseware feature. Would like first round of review feedback by EOD Friday Jan 9, please.
HTML output is here for easier viewing: http://draft-cohorted-courseware.readthedocs.org/en/latest/cohorts/cohorted_courseware.html

lamagnifica commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,224 @@
+
+
+#############################
+Creating Cohorted Courseware
```
Creating Courseware for Student Cohorts

lamagnifica commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,224 @@
+
+You can design your course so that students in one cohort group are given
+different content than students in another cohort group. You do this by
+associating one or more cohort groups with a cohorted content group, and
```
This draft includes many uses of "cohorted". I would revise all of them.

lamagnifica commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,224 @@
+
+.. list-table::
+    :widths: 15 30
+
```
add a line with this instruction:

   :header-rows: 1

Then it will be formatted in bold without you needing to do it, and also get repeated on the next page if the table breaks across pages...

lamagnifica commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,224 @@
+
+Complete these steps to create cohorted content in your course. Optionally, you
+can transpose the order of steps 4 and 5.
+
```
move the "optionally" statement to come after the list? it's kind of confusing to read this caveat before you see the list of things you need to do

lamagnifica commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,224 @@
+Example: Cohorted Courseware
+********************************
+
+You have four cohort groups in your course: A, B, and C. In one subsection in
```
A, B, C, and D? or are there only three groups?

lamagnifica commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,224 @@
+
+To achieve this, on the **Group Configurations** page in Studio, you create two
+cohorted content groups, one called "Video" and the other called "Reading". In
+the Instructor Dashboard, on the **Membership** tab, you associate cohort group
```
I think it may be worth calling out that the instructor dash is in the LMS. You do a step in Studio, you do another step in the LMS, and then you go back to Studio.

lamagnifica commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,224 @@
+A with the "Video" content group, and you associate cohort groups B and C with
+the "Reading" content group. Then, in your course outline, in the relevant
+subsection, you edit the video component so that it is visible only to the
+"Video" content group (which consists of cohort group A) and you edit the
```
I'd change the parenthetical to something more like (now associated with cohort group A). The verb "consists" doesn't seem accurate.

lamagnifica commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,224 @@
+different content than students in another cohort group. You do this by
+associating one or more cohort groups with a cohorted content group, and
+designating specific course components as visible to particular content groups.
+Any course components that do not have an explicitly restricted visibility
```
it's not clear what an "explicitly restricted visibility setting" has to do with content groups and cohort groups

lamagnifica commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,224 @@
+
+You have four cohort groups in your course: A, B, and C. In one subsection in
+your course, you want the students in cohort group A to see a video, while the
+students in the remaining two cohort groups see a reading assignment. You want
```
in the remaining two cohort groups --> in cohort groups B and C to

lamagnifica commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,224 @@
+
+As a final step, in the Instructor Dashboard you preview the course to ensure
+that students see the content that is intended for them. You view the course as
+a member of each of the content groups to confirm that as a student in the
```
content groups or cohort groups?

lamagnifica commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,224 @@
+As a final step, in the Instructor Dashboard you preview the course to ensure
+that students see the content that is intended for them. You view the course as
+a member of each of the content groups to confirm that as a student in the
+"Video" content group, you see a video and the final quiz but not the reading
```
you're testing the experience of the student cohorts, not of the course content, so I'd use cohort group A here, and cohort groups B and C below

catong replied:
Users are selecting the name of the content group to view, so it makes more sense to talk about the view in those terms.

lamagnifica replied:
Personally, I think it's going to be very confusing for them to choose a cohort group in some paces, and a content group in others.

andy-armstrong replied:
@catong @lamagnifica The key thing to get across is that the author is creating the content and doesn't know what the cohorts will be (unless they are also the instructor in the course). The author creates content groups, associates content with those groups, and then previews to verify that the right content is shown for the right groups. All of this can/should happen before any cohorts have been created.

When the course is ready to be run (either directly or as a re-run), then cohorts will be set up in the LMS and associated with content groups as appropriate.

I wonder if the "View This Course As": "student in CONTENT GROUP" wording is now confusing. It makes it sound as if you are going to view the course as if you were a user in a group (aka cohort). In fact, what it is doing is letting you view the course as if you are a student who has access to a particular content group.

lamagnifica commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,224 @@
+******************************
+
+#. In Studio, select **Settings** then select **Group Configurations**.
+#. On the **Group Configurations** page, click **New cohorted content group**.
```
nooooo
not in the UI!

lamagnifica commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,224 @@
+After you create a content group, you can go to the Instructor Dashboard to
+associate it with one or more cohort groups, or work with your course outline to
+specify which components are visible to specific content groups.
+
```
up until now, I had the impression that you could only associate a single component with a content group. <musing>is there a way to diagram this, showing content groups related to course components in the outline in Studio, then showing the result in the LMS?</musing>

srpearce commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,224 @@
+#. :ref:`Create cohorted content groups in Studio<Creating Content Groups>`.
+#. :ref:`Associate one or more cohort groups with a cohorted content
+   group<Associate Cohort Groups with Content Groups>`.
+#. In your course outline, :ref:`Specify Components in Courseware as Visible Only to Certain Content Groups`.
```
For this step and step 6, format links the same way as in earlier steps. Currently, the names of the sections appear in sentence caps (that is, step 5 says "In your course outline, Specify Components in Courseware...").

lamagnifica commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,224 @@
+For details, see :ref:`Associate Cohort Groups with Content Groups` and
+:ref:`Specify Components in Courseware as Visible Only to Certain Content Groups`.
+
+.. note:: Once a content group is created, you cannot delete it. You can
```
Once --> After. rephrase in active? You cannot delete the content groups that you create.

lamagnifica commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,224 @@
+.. note:: Once a content group is created, you cannot delete it. You can
+   remove the association between a cohorted content group and its cohort groups
+   by associating it with another cohort group, or by changing the association
+   to **Not Selected**.
```
or, I suppose, you don't have to assign it to an course content, right?
also, can you rename them?

lamagnifica commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,224 @@
+Associate Cohort Groups with Content Groups
+************************************************
+
+After you have created a content group, you can associate it with one or more
```
have created --> create

lamagnifica commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,224 @@
+After you have created a content group, you can associate it with one or more
+cohort groups. You can use content groups to specify that components in your
+course are visible only to students in the cohort group or groups associated
+with content groups.
```
what are these groups associated with content groups? that sort of comes out of left field.

lamagnifica commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,224 @@
+   If the cohort groups that you want to associate with your content group do
+   not yet exist, you can create them here.
+   
+4. Click the **Settings** tab for the selected cohort group. Under **Associated Cohorted Content Group**, choose the **Select a Content Group** option.
```
noooo

lamagnifica commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,224 @@
+Specify Components in Courseware as Visible Only to Certain Content Groups
+***************************************************************************
+
+After you have created at least one content group, you can edit your course in
```
have created --> create

srpearce commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,224 @@
+Cohorted Content Groups Overview
+********************************
+
+Cohorted content groups, also referred to as content groups, are the means by
```
I'd italicize "cohorted content groups" and "content groups" for emphasis (and add to the glossary)

srpearce commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,224 @@
+that students see the content that is intended for them. You view the course as
+a member of each of the content groups to confirm that as a student in the
+"Video" content group, you see a video and the final quiz but not the reading
+assignment; and as a student in the "Reading" content group, you see a reading
```
change semicolon to comma

lamagnifica commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,224 @@
+
+After you have created at least one content group, you can edit your course in
+Studio and designate whether all students, or only particular content groups,
+can see certain components.
```
I think this will work better if the content groups are just organizational structures, while cohort groups represent a set of human beings. That is, the content group doesn't have the ability to see things. It is just a container, or perhaps a means  of tagging, the course content. Maybe,
After you create a content group, you identify the components that you want to include in that group. Any component  that you do not tag as being visible to a specified content group remains visible to all enrolled students.

andy-armstrong replied:
+1 @lamagnifica

lamagnifica commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,224 @@
+.. note:: You do not need to edit the visibility settings of components that are
+   intended for all students. Components that you do not explicitly indicate as
+   visible to a group are visible to all students enrolled in your course,
+   regardless of the cohort group that they belong to.
```
right, I'd put this up front

catong commented:
@explorerleslie @andy-armstrong @cahrens @dan-f @jimabramson @talbs
@srpearce @lamagnifica @mhoeber 
Sorry for the spam! This PR is updated with the terminology changes:
"cohorted content group" -> "content group" and "cohort group" -> "cohort".
(Please note that not all links were updated so if you are reviewing the raw file please ignore link text that might still contain the term "cohort group" -- this is not the visible text in HTML or PDF output.)

Updated draft HTML output is here and will be easiest to review:
http://draft-cohorted-courseware.readthedocs.org/en/latest/cohorts/index.html

andy-armstrong commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,223 @@
+You can design your course so that students in one cohort are given different
+content than students in another cohort. You do this by associating one or more
+cohorts with a content group, and designating specific course components as
+visible to particular content groups. Any course components that do not have an
```
I suggest adding a one sentence description of what a content group is here. I think this section will be too confusing to read in order if you need to go to "About Content Groups" just to understand the terminology.

andy-armstrong commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,223 @@
+regardless of their cohort. For more details about content groups, see
+:ref:`About Content Groups`.
+
+Complete these steps to create cohorted content in your course. Optionally, you
```
I think it would be worth having an explanation of the typical workflow here. This is how I think about it (@explorerleslie may have different use cases):
1. The author creates a course with content groups to designate certain content as being only shown to certain cohorts.
2. The author tests their course using preview to verify that all the content works as expected.
3. The course is published either directly or as a re-run
4. Once the course is ready in LMS, the instructor creates the cohorts that they desire for this particular run of the course. When they create the cohort is likely to be the time they decide which content group is appropriate for them (although they can change it later)
5. The instructor should then test the course using test users in each cohort to verify that the behavior is as they intended.

For me, I find it confusing to mix the author's job (steps 1 and 2) with the instructor's job (steps 3 through 5). I'd like to see this documentation follow this kind of a flow, rather than creating cohorts before creating content groups.

explorerleslie replied:
Totally absolutely agree with @andy-armstrong 's comments above. would like to see the author and course staff roles and workflows split out here. I think it will clarify things a lot.

andy-armstrong commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,223 @@
+Example: Cohorted Courseware
+********************************
+
+You have three cohorts in your course: A, B, and C. In one subsection in your
```
I'd prefer to have more realistic groups than just A, B and C. Can we come up with a typical scenario that would make sense for most readers of this document?

dan-f replied:
On a different note, I found the declarative nature of the first sentence confusing. We don't want to convince the user that they actually have a course with cohorts A, B, and C. Could it say "Suppose You have three cohorts in your course: A, B, and C."?

explorerleslie replied:
+1 on both comments above.

explorerleslie replied:
I would say "In one unit" instead of "in one subsection". You can only mark content as visible to only a specific content group at the component level. You cannot mark whole units, subsections, or sections as only visible to a specific content group. I think it would be good to call this out explicitly. We have people trying to hack content experiments to support experiments on whole units/sections/subsections and I would be surprised if the same thing happens here.

andy-armstrong commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,223 @@
+component, because if no content group is specified in a component's visibility
+settings, it is visible to all students.
+
+As a final step, in the Instructor Dashboard you preview the course to ensure
```
Typo: you don't preview in the Instructor Dashboard...

andy-armstrong commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,223 @@
+Associate Cohorts with Content Groups
+*************************************
+
+After you have created a content group, you can associate it with one or more
```
As mentioned before, I'd like to separate the work done in Studio from that done in LMS. I think it would be better to move this section to after the one about specifying content groups for components.

andy-armstrong commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,223 @@
+You can associate additional cohorts with the same or a different content group
+by repeating steps 3 to 6.
+
+For an example of using content groups to create cohorted courseware, see
```
Is it worth saying something here about how you test your cohort? Currently the instructor's only option is to create test users and manually put them into each cohort, and to then log in as that user.

andy-armstrong commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,223 @@
+.. _Specify Components in Courseware as Visible Only to Certain Content Groups:
+
+***************************************************************************
+Specify Components in Courseware as Visible Only to Certain Content Groups
```
I think this title doesn't work with the new sense of content groups not truly being groups. It isn't that they are visible only to certain content groups, because content groups aren't entities that see things. A user doesn't belong to a content group, but instead belong to a cohort that has privileged access to a content group.

andy-armstrong commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,223 @@
+.. _View the Course as a Member of a Content Group:
+
+***********************************************
+View the Course as a Member of a Content Group
```
As mentioned previously, I find it confusing to say that a user is a member of a content group. Having said that, maybe this is the way we have to present things if we are to call them 'content groups'. The user can think of the content group as being a virtual group that represents all the users who are in any of the cohorts that have been associated with that content group. This does fit quite well with experiment groups which are also virtual in the sense that users are dynamically assigned into them. Maybe I'm coming around to this naming convention... :-)

explorerleslie replied:
I think we should leave it as is for now, it's an easier mental model for the user to think about users as part of a particular content group. They are, in fact, members of a particular content group, albeit through the cohort, instead of directly being assigned.

andy-armstrong commented:
@catong I've completed my review of this draft. It is looking excellent! My major challenge is around the terminology as we are all experiencing. I would also like to separate the authoring workflow from the instructor workflow, but I'd run that by @explorerleslie to see if I'm off base here.

dan-f commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,223 @@
+#. :ref:`Associate one or more cohorts with a content group<Associate Cohorts
+   with Content Groups>`.
+   
+#. In your course outline, :ref:`Specify Components in Courseware as Visible
```
It looks odd to me that the whole sentence isn't a link. Maybe because it breaks the pattern?

dan-f commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,223 @@
+To achieve this, on the **Group Configurations** page in Studio, you create two
+content groups, one called "Video" and the other called "Reading". In the
+Instructor Dashboard, on the **Membership** tab, you associate cohort A with the
+"Video" content group, and you associate cohorts B and C with the "Reading"
```
I'm worried that naming this content group "Video" might teach readers the false notion that content groups are only for one component or subsection. Rather than make the example more complicated, perhaps you could add something describing how you can use the same content groups for content that spans across sections and subsections?

dan-f commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,223 @@
+
+***********************
+Content Groups Overview
+***********************
```
This would be helpful to see before the example, since someone new to cohorted courseware might wonder why we need content groups at all. They would probably wonder why they can't just choose the cohort directly in studio. I think the many-to-many content group <-> cohort relationship is the key.

mhoeber replied:
agree. Coming to this new, i didn't get this at first.

dan-f commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,223 @@
+#. From the drop down list, select the cohort that you want to associate
+   with your content group.
+
+   If the cohorts that you want to associate with your content group do not yet
```
Something about the spacing in the rendered version here looks a little funky to me.

mhoeber replied:
@dan-f  our great new template will take care of this.  saudi doc we be rebased to use the template before the 15th.
http://edx.readthedocs.org/projects/edx-partner-course-staff/en/latest/

dan-f commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,223 @@
+      - You see any content that is intended for all
+        students.
+    * - Student in <Content Group Name>            
+      - You see content that is intended for all students, as well
```
The horizontal scrolling here is inconvenient. Wasn't there a similar issue in a prior doc PR? What was decided in that case?

lamagnifica replied:
We are in the midst of rolling out a new template for our HTML docs, @dan-f . Table formatting -- lines now wrap -- is my favorite of the many fixes that Mark has made.

dan-f commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,223 @@
+    * - Staff
+      - You see all content in the course, including content
+        that is hidden from students.
+    * - Student
```
I wonder if we should change this from "Student" to "All Students" (in both UI and doc), since that's what we call it in Studio.

mhoeber commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,223 @@
+Instructor Dashboard, on the **Membership** tab, you associate cohort A with the
+"Video" content group, and you associate cohorts B and C with the "Reading"
+content group. Then, in your course outline, in the relevant subsection, you
+edit the video component so that it is visible only to the "Video" content group
```
the components are on the unit page.  is it accurate to say "in the relevant unit ..."

mhoeber commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,223 @@
+#. Repeat this step to create as many content groups as you want.
+
+After you create a content group, you can go to the Instructor Dashboard to
+associate it with one or more cohorts, or work with your course outline to
```
it's not really "or", right?  Ultimately to make it all work you have to both specify component visibility and associate with cohorts.  A way to make clear this workflow?

mhoeber commented:
@catong  as you can set content groups on discussion components, how does this relate to new ability to control whether or not content discussions are always cohorted?

mhoeber commented:
Sure you know just as reminder, we'll need to update: http://edx.readthedocs.org/projects/edx-partner-course-staff/en/latest/developing_course/course_components.html and http://edx.readthedocs.org/projects/edx-partner-course-staff/en/latest/developing_course/testing_courseware.html

and pretty much everywhere there's studio components (let me know if you need help with other random screen shots.)

andy-armstrong commented:
@mhoeber Really great question about discussion handling. We have a story to work it out that won't land for 1/15:

https://openedx.atlassian.net/browse/TNL-650

explorerleslie commented:
@catong for section 10.3.1. Example: Cohorted Courseware: I'd like to see a more realistic example here. One example we used in the discussion cohorts doc is around alumni cohorts. You could present the use case of University Alumni cohort and Current University Students cohort being created, and both of them are associated with the University Content content group. Then, course authors can give special video messages from the dean/president/whatever of the college at the end of every section that are only applicable or interesting to students associated with the University in those two cohorts. Let me know if you want to chat through the example further.

explorerleslie commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,223 @@
+Content Groups Overview
+***********************
+
+Content groups are the means by which you can designate specific course content
```
total style nit: I hate the phrase "the means by which you", but you may love it, so feel free to ignore. :) I normally do "Content groups enable authors to designate specific course content as visible to particular cohorts of students." Or "With content groups, authors can designate specific course content as visible to particular cohorts of students.

explorerleslie commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,223 @@
+#. Enter a meaningful name for the content group, then click **Create**.
+#. Repeat this step to create as many content groups as you want.
+
+After you create a content group, you can go to the Instructor Dashboard to
```
I would rip this out of here. I really want to delineate one clear workflow for authoring, and then jump back to the course staff workflow.

explorerleslie commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,223 @@
+#. In Studio, select **Content**, then select **Outline**.
+
+#. For each component that you want to make visible only to a particular content
+   group or groups, click the **Visibility Settings** icon.
```
Do we have any precedent for showing the icon the first time we introduce it in the doc? I don't know if I would know that the eyeball was the Visibility Settings icon. Is that the tooltip on the icon? (I don't have a sandbox to check right now.)

catong replied:
This pass does not includes screenshots. I was definitely planning on adding at least one of the eyeball icon, and that should make it clearer.

explorerleslie commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,223 @@
+   visible to a group are visible to all students enrolled in your course,
+   regardless of the cohort that they belong to.
+
+#. In Studio, select **Content**, then select **Outline**.
```
You're missing a step after this. After you go to the outline, you have to click on a specific unit to open up the view where you will see all components for that unit, and then you'll be able to click the Visibility Settings icon.

explorerleslie commented:
@catong I think we should add a mention in section 6.7. Controlling Content Visibility, linking to the new cohort-specific courseware section. 6.7 feels incomplete if we don't mention it.

explorerleslie commented:
@catong The screenshots and content in section 6.8. Testing Your Course Content will need to be updated for the new preview workflow.

explorerleslie commented:
@catong I also think we should link from 6.8 Testing Your Course Content to 10.3.6. View the Course as a Member of a Content Group.

explorerleslie commented:
@catong completed a review and spewed comments all over the place. great work so far! Let me know when you want another review.

catong commented:
@andy-armstrong @dan-f @explorerleslie @lamagnifica @mhoeber @srpearce thanks for the great review comments. Next pass will include the suggested LMS/Studio workflow separation, clearer workflow, screenshots, plus the impacts on the other sections which I was leaving out of the PR for the first pass. Thanks!

catong commented:
@explorerleslie @andy-armstrong @dan-f @lamagnifica @mhoeber @srpearce 
Please review this 2nd pass. Changes in the latest commit include incorporating review comments plus:
Added brief section for content groups, added screenshots in testing_courseware.rst to reflect addition of visibility settings for content groups and changes in the preview button design
Add screenshots in cohorted_courseware.rst
Add mention of visibility for content groups in controlling_content_visibliity.rst
Add mention of visibility of content groups in workflow.rst

andy-armstrong commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,280 @@
+.. _Preview Cohort Specific Courseware:
+
+*************************************
+Previewing Cohort-Specific Courseware
```
I think this should be "Preview cohort-specific courseware" as otherwise it doesn't match the other sections. You can see this in the index at the beginning:

    Assign students to cohorts.
    Associate one or more cohorts with a content group.
    Previewing Cohort-Specific Courseware

andy-armstrong commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,280 @@
+#. :ref:`Assign students to cohorts<Options for Assigning Students to Cohorts>`.  
+#. :ref:`Associate one or more cohorts with a content group<Associate Cohorts
+   with Content Groups>`.
+#. :ref:`Preview Cohort Specific Courseware`.
```
I think this should be "Preview cohort-specific courseware" as otherwise it doesn't match the other sections. You can see this in the index at the beginning:

    Assign students to cohorts.
    Associate one or more cohorts with a content group.
    Previewing Cohort-Specific Courseware

andy-armstrong commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,280 @@
+
+#. :ref:`Enable cohorts in your course<Enabling and Configuring Cohorts>`.
+#. :ref:`Create content groups<Creating Content Groups>`.
+#. :ref:`Specify Components in Courseware as Visible Only to Certain Content
```
This third item needs a different version for the list to match the previous two.

andy-armstrong commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,280 @@
+Example: Cohort-Specific Courseware
+***********************************
+
+Suppose that you have two cohorts in your course: University Alumni and Current
```
I found the introduction to this example confusing. I think it only makes sense if there are three cohorts: alumni, current students and the default cohort (everybody else). Note that the default cohort is only created as needed, so you cannot assume that all courses will have one.

andy-armstrong commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,280 @@
+cohorts. 
+
+You can design your course so that some students are given different content
+than others. You do this by creating content groups in Studio, and designating
```
IMO "content groups" still need some kind of introduction. Maybe just putting it in quotes would be enough to indicate that this isn't a term that you should already be familiar with. If not, then a sentence saying what a content group is would help.

catong replied:
Another thing I've been wrestling with... what is a content group other than a container? which seems to me not something we want/need to say to users. So I've been trying to explain what it is by explaining what you do with it. Will try to come up with something.

andy-armstrong replied:
I don't think you should describe it as a container. It doesn't really contain anything. I'm thinking of it now as a virtual/conceptual group of students who should be shown a particular set of content. The author shouldn't have to think about exactly which students these are because that will be configured by the course team when the course is run.

andy-armstrong commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,280 @@
+
+Use content groups to designate specific course content as visible to particular
+cohorts of students. You create content groups and associate them with one or
+more cohorts. Then, in your course outline, for any components that you want to
```
This is pedantic, but you can have content groups that are not associated with any cohorts. For example, you might have a run of a course where there is no need to show content associated with a particular content group.

Also, I feel that you associate cohorts with content groups, not the other way around. It is weird to say that you associate content groups with cohorts when you actually can't do that in the UI. You have to create the content group first, and then when you are managing a cohort you can associate a content group with it.

catong replied:
I agree -- the reason I ended up with this version was because in this case I wanted to keep the focus on content groups. I had been wrestling with various ways of describing "associate cohorts with content groups" here, and thought it would be more confusing to switch the focus to cohorts. Let me work on this!

andy-armstrong commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,280 @@
+      - You see content that is intended for all students, as well
+        as any content specifically set to be visible to this content group.
+
+#. In Studio, in the course outline, click **Preview Changes**. You see your
```
I wonder if you want to mention that you can also view your course as a student with "View Live". For example, on the day that a new section goes live, the author might wish to go in and verify that the live course is correctly showing cohorted courseware to the correct content groups.

catong replied:
Good point, thanks.

catong commented:
@andy-armstrong @explorerleslie I've made further updates based on excellent review comments from Andy. 
HTML is updated: http://draft-cohorted-courseware.readthedocs.org/en/latest/cohorts/cohorted_courseware.html

andy-armstrong commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,298 @@
+
+Content groups do not have an actual impact on the visibility of a course
+component until they are associated with one or more cohorts. If you have
+designated certain course content as only visible to a content groups, cohorts
```
Typo: visible to a content groups

andy-armstrong commented:
:+1: I think this is excellent, Carol! Nice job describing such a conceptually complicated feature!

Don't forget that you'll need to update your screenshots *again* once my doc string PR lands.

lamagnifica commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,298 @@
+Example`.
+
+
+Complete these steps to create cohorted content in your course. 
```
cohorted content in your course --> cohort-specific course content

lamagnifica commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,298 @@
+designated certain course content as only visible to a content groups, cohorts
+that are associated with that content group will see the designated content.
+
+For an example of using content groups to create cohorted courseware, see
```
...to create cohort-specific content for your course,...

lamagnifica commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,298 @@
+test knowledge of the concepts taught in that section. The quiz will be shown to
+all students enrolled in the course.
+
+To achieve this, on the **Group Configurations** page in Studio, you create one
```
strike the second comma

lamagnifica commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,298 @@
+group.
+
+Then, in your course outline, you change the visibility settings for the video
+component at the end of each section so that so that it is visible only to the
```
strike duplicated "so that"

lamagnifica commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,298 @@
+
+As a final step, you preview the course in the LMS to ensure that students see
+the content that is intended for them. You confirm that when you view the course
+in the role of **Student**, in other words, any student not in a content group,
```
actual parentheses would make this easier to parse:
You confirm that when you view the course in the role of **Student** (in other words, any student not in a content group) you see a quiz...

lamagnifica commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,298 @@
+in the role of **Student**, in other words, any student not in a content group,
+you see a quiz at the end of each section but not the university-related videos,
+and when you view the course as a student in the "University-Specific Content"
+group, you see a university- related video as well as the quiz at the end of
```
delete extra space after hyphen

lamagnifica commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,298 @@
+To achieve this, on the **Group Configurations** page in Studio, you create one
+content group called "University-Specific Content". In the Instructor Dashboard,
+on the **Membership** tab, you associate both the "University Alumni" and the
+"Current University Students" cohorts with the "University Content" content
```
""University Content" content" --> "University-Specific Content" content

lamagnifica commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,298 @@
+
+***********************************
+Example: Cohort-Specific Courseware
+***********************************
```
this is soooo helpful. terrific

lamagnifica commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,298 @@
+Content Groups
+**************
+
+Content groups are virtual groupings of students who will see a particular set
```
I'm still squeamish about characterizing a content group as a collection of students. That's what a cohort is. I think the two concepts become harder to get across when the distinction is fuzzy...

lamagnifica commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,298 @@
+of course content. You can use content groups to designate specific course
+content as visible to particular :ref:`cohorts<Cohorts Overview>` of students. 
+
+You create content groups in Studio, and in your course outline, you use the
```
could break this into two sentences, or strike the second comma

lamagnifica commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,298 @@
+regardless of their cohort.
+
+Content groups do not have an actual impact on the visibility of a course
+component until they are associated with one or more cohorts. If you have
```
if --> Only after 
(and then strike "only" in line 106)
??

lamagnifica commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,298 @@
+ 
+#. On the **Group Configurations** page, click **New content group**. If you
+   have not created any content groups, click **Add your first content
+   group**.
```
is there a different label after you create your first content group? not clear what the person creating the 2nd content group in a course (who may not have created the 1st one) has to do...

lamagnifica commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,298 @@
+
+3. Enter a meaningful name for the content group, then click **Create**.
+   The page refreshes to show the name of your new content group.
+#. Repeat this step to create as many content groups as you want.
```
actually this is where any changed label oddness might come into play...

lamagnifica commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,298 @@
+
+After you create a content group, you can work with your course outline to
+specify which components are visible to specific content groups. For details,
+see :ref:`Associate Cohorts with Content Groups`.
```
this link doesn't seem to be directly related to the preceding sentence.. I think this one and the one on lines 138-139 may have gotten switched?

catong replied:
Thanks for catching this!

lamagnifica commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,298 @@
+For details about previewing your course to ensure that students in a cohort
+correctly see the content intended for them, see :ref:`Preview Cohort Specific
+Courseware`.
+
```
here, or maybe in intro statement above, it is probably worth explicitly saying that this particular restriction comes last when the system determines whether a given student can view content: hide from students has to be off, it has to be published, it has to be on or after the release date, and only then is the student in the right cohort. (I think I got the sequence right!)

lamagnifica commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,298 @@
+*************************************
+
+After you create a content group, you can associate it with one or more cohorts
+that should share the same visibility settings for special content in your
```
can you strike "should"?

lamagnifica commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,298 @@
+   
+#. Scroll to the **Cohort Management** section at the bottom.
+
+#. From the drop down list, select the cohort that you want to associate
```
drop-down per Sylvia's style guide

lamagnifica commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,298 @@
+   with your content group.
+
+   If the cohort that you want to associate with your content group does not yet
+   exist, you can create it here.
```
erm, this opens up a lot of potential to add warning: don't add cohorts after your course begins, what you create here is a cohort that you have to assign students to manually... is this sentence really needed?

lamagnifica commented on en_us/course_authors/source/cohorts/cohort_config.rst:
```diff
@@ -244,31 +241,32 @@ and download student data`.
+
+  For example, if your .csv file contains conflicting information such as
+  Student A being first assigned to Cohort 1, then later in the spreadsheet
+  being assigned toCohort 2, the end result of your .csv upload is that Student
```
add space: to^ Cohort 2

lamagnifica commented on en_us/course_authors/source/cohorts/cohort_config.rst:
```diff
@@ -244,31 +241,32 @@ and download student data`.
+  For example, if your .csv file contains conflicting information such as
+  Student A being first assigned to Cohort 1, then later in the spreadsheet
+  being assigned toCohort 2, the end result of your .csv upload is that Student
+  A is assignedto Cohort 2. However, the upload results file will count Student
```
assigned^ to

lamagnifica commented on en_us/course_authors/source/cohorts/cohort_config.rst:
```diff
@@ -244,31 +241,32 @@ and download student data`.
+  Student A being first assigned to Cohort 1, then later in the spreadsheet
+  being assigned toCohort 2, the end result of your .csv upload is that Student
+  A is assignedto Cohort 2. However, the upload results file will count Student
+  A twicein the "Students Added" count: once when they are added to Cohort 1,
```
twice^ in

lamagnifica commented:
you did a phenomenal rewrite in a short time. some cleanup items noted

explorerleslie commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,298 @@
+Suppose that you create two :ref:`cohorts<Cohorts Overview>` in your course: University Alumni and
+Current University Students. Students who are not in either of these cohorts are
+automatically placed into a third cohort, the default cohort, when they access
+the courseware or discussion tabs in the course. For more information about
```
Nit: the item "when they access the courseware or discussion tabs in the course: might be too low-level for this type of example overview.

catong replied:
@explorerleslie, I don't want to remove this entirely as I think the question will be raised as to when/how students are added to the default cohort.

explorerleslie commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,298 @@
+
+.. note:: Once a content group is created, you cannot delete it. You can
+   remove the association between a content group and its cohorts by associating
+   it with another cohort, or by changing the association to **Not Selected**.
```
If you wanted to delete a content group, you wouldn't assign it to another cohort. I would just say You can remove the association between a content group and its cohorts by changing their associations to **Not Selected**."

explorerleslie commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,298 @@
+  :width: 400
+  :alt: Screen capture of unit in course outline with visibility setting icon highlighted 
+
+4. Click **Save**.
```
In the HTML doc, this line appears smashed up against the screenshot. Do we add new lines between?

lamagnifica replied:
@explorerleslie our new template has better white space. (Currently, the new template is used to generate the "real" B&R guide on readthedocs, but we haven't had a chance to reconfigure all of our other drafts to use it)

explorerleslie replied:
got it, thanks @lamagnifica

explorerleslie commented on en_us/course_authors/source/cohorts/cohorted_courseware.rst:
```diff
@@ -0,0 +1,298 @@
+#. From the drop down list, select the content group that you want your cohort
+   to be associated with.
+
+.. image:: ../Images/Cohorts_AssociateWithContentGroup.png
```
Can you retake that screenshot? The styling was borked when you took it, but it's fixed now.

andy-armstrong replied:
I recommend not taking any new screenshots until we've got everything merged and onto a new sandbox. There are a lot of subtle changes going in!

explorerleslie commented:
@catong FABULOUS! this version is so good. Minor comments made, but thanks for the hard work on restructuring -- it made a HUGE difference in the clarity of this doc. Thanks!

explorerleslie commented:
:+1:

catong commented:
Closing this PR to create a new one to merge to Saudi-specific branch.
